### PR TITLE
add metrics for provisioner usage

### LINKED ISSUE #840: Emit Resource Limits in Metrics Controller
**Tell us about your request**
Emit metrics for resource usage by the provisioner.

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
There's a new counter controller that this PR introduces - https://github.com/awslabs/karpenter/pull/817
It sets the resources being consumed by the provisioner in the provisioner.status.Resources but we should also emit this and create a Grafana dashboard off those metrics.

**Are you currently working around this issue?**
N/A

**Additional context**
N/A

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by tzneal**
**1. Issue, if available:**

Fixes #840

**2. Description of changes:**

This adds three metrics for provisioners
- Limits
- Usage
- Usage Percentage (0-100)

**3. How was this change tested?**

Deploying to EKS

**4. Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tzneal** [comment]
Screen Shot 2022-05-31 at 9 39 54 PM

File: Makefile [hunk 1]
```diff
@@ -78,6 +78,9 @@ codegen: ## Generate code. Must be run if changes are made to ./pkg/apis/...
 		output:crd:artifacts:config=charts/karpenter/crds
 	hack/boilerplate.sh
 
+docgen: ## Generate docs
```
Comment (bwagner5) [inline Makefile:81]: can you update the .PHONY list. it looks like we're missing a few of the other ones like "nightly", "snapshot", and "stablerelease" as well.

**Review by bwagner5 — COMMENTED** [review]
nice!

File: hack/docs/metrics_gen_docs.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+				continue
+			}
+			packages = append(packages, pkg)
+		}
+		return nil
+	})
+
+	// metrics are all package global variables
+	var allMetrics []metricInfo
+	for _, pkg := range packages {
```
[the remark this answers is not in the document]
Comment (tzneal) [reply hack/docs/metrics_gen_docs.go:82]: Good catch, fixed.

**Review by bwagner5 — APPROVED** [review]
lgtm

File: hack/docs/metrics_gen_docs.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+description: >
+  Inspect Karpenter Metrics
+---
+`)
+	fmt.Fprintf(f, "<!-- this document is generated from hack/docs/metrics_gen_docs.go -->\n")
+	fmt.Fprintf(f, "Karpenter writes several metrics to Prometheus to allow monitoring cluster provisioning status\n")
+	previousSubsystem := ""
+	for _, metric := range allMetrics {
+		if metric.subsystem != previousSubsystem {
+			fmt.Fprintf(f, "## %s%s Metrics\n", strings.ToTitle(metric.subsystem[0:1]), metric.subsystem[1:])
```
[the remark this answers is not in the document]
Comment (tzneal) [reply hack/docs/metrics_gen_docs.go:121]: No, you can see it here: https://deploy-preview-1872--karpenter-docs-prod.netlify.app/preview/tasks/metrics/

**Pull request merged**